luxtronik/discover.py

```diff
@@ -0,0 +1,56 @@
+"""Discovery funtion for Luxtronik2 heatpump controllers."""
+import logging
+import socket
+
+LOGGER = logging.getLogger("Luxtronik.Discover")
+
+
+def discover() -> list[(str, int)]:
+    """Broadcast discovery for luxtronik heatpumps."""
+
+    results: list[(str, int)] = list()
+
+    for magic_port in (4444, 47808):
+        LOGGER.debug("Send discovery packets to port %s", magic_port)
+        server = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
+        server.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
+        server.bind(("", magic_port))
+        server.settimeout(2)
```

Comment:
The timeout value should be put into a constant also, so it could be better understood. Suggestion: `LUXTRONIK_DISCOVERY_TIMEOUT`